src/data/text/item_descriptions.h

```diff
@@ -743,10 +743,17 @@ static const u8 sSootheBellDesc[] = _(
     "calms spirits and\n"
     "fosters friendship.");
 
+#if B_MENTAL_HERB < GEN_5
 static const u8 sMentalHerbDesc[] = _(
     "A hold item that\n"
     "snaps POKéMON out\n"
     "of infatuation.");
+#else
+static const u8 sMentalHerbDesc[] = _(
+    "Snaps Pokémon out\n"
+    "of move-binding\n"
+    "effects.");
+#endif
```

Comment:
I would instead add this change alongside a check for `BATTLE_ENGINE` define to Item Expansion since we don't update item data in Battle Engine Upgrade.